app/src/main/java/com/ecosystem/kin/app/MainActivity.java

```diff
@@ -201,6 +212,16 @@ private void createNativeSpendOffer() {
         }
     }
 
+    private void createNativeEarnOffer() {
+        String userID = SignInRepo.getUserId(getApplicationContext());
+        String offerJwt = JwtUtil.generateEarnOfferExampleJWT(BuildConfig.SAMPLE_APP_ID, userID);
+        try {
+            Kin.requestPayment(offerJwt, getNativeEarnOrderConfirmationCallback());
```

Comment:
add away to cancel callback - for example when activity destroyed also on native spend (open task for that)